error: macro "CV_Assert" passed 3 arguments, but takes just 1

##### System information (version)


- OpenCV => 3.1 (a freshly cloned git repository)
- Operating System / Platform =>  Ubuntu 16.04 x64
- Compiler => gcc 7.3.0

##### Detailed description

When I try to run `make`, the output is the following:

```
[ 33%] Linking CXX shared library ../../lib/libopencv_dnn.so
[ 33%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_dc1394_v2.cpp.o
[ 33%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_gstreamer.cpp.o
[ 33%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_v4l.cpp.o
[ 33%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_ffmpeg.cpp.o
[ 33%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_gphoto2.cpp.o
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:110:43: error: macro "CV_Assert" passed 3 arguments, but takes just 1
         CV_Assert(!xine, !stream, !vo_port);
                                           ^
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:210:40: error: macro "CV_Assert" passed 3 arguments, but takes just 1
         CV_Assert(xine, vo_port, stream);
                                        ^
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:243:40: error: macro "CV_Assert" passed 3 arguments, but takes just 1
         CV_Assert(xine, vo_port, stream);
                                        ^
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:293:40: error: macro "CV_Assert" passed 3 arguments, but takes just 1
         CV_Assert(xine, vo_port, stream);
                                        ^
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:312:40: error: macro "CV_Assert" passed 3 arguments, but takes just 1
         CV_Assert(xine, vo_port, stream);
                                        ^
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:331:40: error: macro "CV_Assert" passed 3 arguments, but takes just 1
         CV_Assert(xine, vo_port, stream);
                                        ^
[ 33%] Built target opencv_dnn
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp: In member function ‘bool XINECapture::open(const char*)’:
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:110:9: error: ‘CV_Assert’ was not declared in this scope
         CV_Assert(!xine, !stream, !vo_port);
         ^~~~~~~~~
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:110:9: note: suggested alternative: ‘CV_StsAssert’
         CV_Assert(!xine, !stream, !vo_port);
         ^~~~~~~~~
         CV_StsAssert
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp: In member function ‘virtual double XINECapture::getProperty(int) const’:
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:210:9: error: ‘CV_Assert’ was not declared in this scope
         CV_Assert(xine, vo_port, stream);
         ^~~~~~~~~
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:210:9: note: suggested alternative: ‘CV_StsAssert’
         CV_Assert(xine, vo_port, stream);
         ^~~~~~~~~
         CV_StsAssert
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp: In member function ‘bool XINECapture::oldSeekFrame(int)’:
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:243:9: error: ‘CV_Assert’ was not declared in this scope
         CV_Assert(xine, vo_port, stream);
         ^~~~~~~~~
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:243:9: note: suggested alternative: ‘CV_StsAssert’
         CV_Assert(xine, vo_port, stream);
         ^~~~~~~~~
         CV_StsAssert
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp: In member function ‘bool XINECapture::seekFrame(int)’:
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:293:9: error: ‘CV_Assert’ was not declared in this scope
         CV_Assert(xine, vo_port, stream);
         ^~~~~~~~~
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:293:9: note: suggested alternative: ‘CV_StsAssert’
         CV_Assert(xine, vo_port, stream);
         ^~~~~~~~~
         CV_StsAssert
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp: In member function ‘bool XINECapture::seekTime(int)’:
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:312:9: error: ‘CV_Assert’ was not declared in this scope
         CV_Assert(xine, vo_port, stream);
         ^~~~~~~~~
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:312:9: note: suggested alternative: ‘CV_StsAssert’
         CV_Assert(xine, vo_port, stream);
         ^~~~~~~~~
         CV_StsAssert
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp: In member function ‘bool XINECapture::seekRatio(double)’:
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:331:9: error: ‘CV_Assert’ was not declared in this scope
         CV_Assert(xine, vo_port, stream);
         ^~~~~~~~~
/home/enoch/opencv/modules/videoio/src/cap_xine.cpp:331:9: note: suggested alternative: ‘CV_StsAssert’
         CV_Assert(xine, vo_port, stream);
         ^~~~~~~~~
         CV_StsAssert
modules/videoio/CMakeFiles/opencv_videoio.dir/build.make:230: recipe for target 'modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_xine.cpp.o' failed
make[2]: *** [modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_xine.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:2664: recipe for target 'modules/videoio/CMakeFiles/opencv_videoio.dir/all' failed
make[1]: *** [modules/videoio/CMakeFiles/opencv_videoio.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/modules/videoio/src/cap_xine.cpp
+++ b/modules/videoio/src/cap_xine.cpp
@@ -107,7 +107,7 @@ class XINECapture : public IVideoCapture
 
     bool open(const char *filename)
     {
-        CV_Assert(!xine, !stream, !vo_port);
+        CV_Assert_N(!xine, !stream, !vo_port);
         char configfile[2048] = {0};
 
         xine = xine_new();
@@ -207,7 +207,7 @@ class XINECapture : public IVideoCapture
 
     double getProperty(int property_id) const CV_OVERRIDE
     {
-        CV_Assert(xine, vo_port, stream);
+        CV_Assert_N(xine, vo_port, stream);
         int pos_t, pos_l, length;
         bool res = (bool)xine_get_pos_length(stream, &pos_l, &pos_t, &length);
 
@@ -240,7 +240,7 @@ class XINECapture : public IVideoCapture
 protected:
     bool oldSeekFrame(int f)
     {
-        CV_Assert(xine, vo_port, stream);
+        CV_Assert_N(xine, vo_port, stream);
         // no need to seek if we are already there...
         if (f == frame_number)
         {
@@ -290,7 +290,7 @@ protected:
 
     bool seekFrame(int f)
     {
-        CV_Assert(xine, vo_port, stream);
+        CV_Assert_N(xine, vo_port, stream);
         if (seekable)
         {
             int new_time = (int)((f + 1) * (float)frame_duration);
@@ -309,7 +309,7 @@ protected:
 
     bool seekTime(int t)
     {
-        CV_Assert(xine, vo_port, stream);
+        CV_Assert_N(xine, vo_port, stream);
         if (seekable)
         {
             if (xine_play(stream, 0, t))
@@ -328,7 +328,7 @@ protected:
 
     bool seekRatio(double ratio)
     {
-        CV_Assert(xine, vo_port, stream);
+        CV_Assert_N(xine, vo_port, stream);
         if (ratio > 1 || ratio < 0)
             return false;
         if (seekable)

